hangman.py

```diff
@@ -1,213 +1,306 @@
+    |  _  |/ _` | '_ \ / _` | '_ ` _ \ / _` | '_ \   | | |  /| |
+    | | | | (_| | | | | (_| | | | | | | (_| | | | | _| |_\ |_/ /
+    \_| |_/\__,_|_| |_|\__, |_| |_| |_|\__,_|_| |_| \___(_)___/ 
+                        __/ |                                   
+                       |___/                                    
+    ''')
+    print(f'I am thinking of a word that is {len(secret_word)} letters long.')
+
+
+def hangman(secret_word, hints=False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    welcome_art(secret_word)
+
+    # start variables
+    guesses = 6
+    letters_guessed = []
+    vowels = ('a', 'e', 'i', 'o', 'u')
+    warnings = 3
+    style = TextStyle()
+
+    #main game cicle
+    while 1:
+        if is_word_guessed(secret_word, letters_guessed):
+            victory(guesses, secret_word)
+            break
+        elif guesses <= 0:
+            style.change_background_font(f'You lost. Guesses exceeded. The secret word was - {secret_word}')
+            break
+        else:
+            # let the user know his resources
+            print(get_guessed_word(secret_word, letters_guessed))
+            print(f'You have {guesses} guesses left.')
+            available = get_available_letters(letters_guessed)
+            print(f'Available letters: {available}')
+
+            # validate letter
+            l_c = letter_check(guesses, warnings)
+            letter, guesses, warnings = l_c[0], l_c[1], l_c[2]
+            letters_guessed.append(letter.lower())
+
+            # check if user want hint
+            if hints:
+                if letter == '*':
+                    print('Possible word matches are: ', end=' ')
+                    show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+
+            # check if letter in the word and make corresponding output
+            if letter.lower() in secret_word:
+                style.bold('Right guess!')
+
+            elif letter.lower() in vowels and letter.lower() not in secret_word:
+                style.red('Wrong guess(')
+                guesses -= 2
+            else:
+                style.red('Wrong guess(')

```

Comment:
Could you make the output to be identical to the examples in requirements?

Reply:
Done